src/openforms/registrations/tasks.py

```diff
@@ -54,9 +63,15 @@ def register_submission(submission_id: int) -> Optional[dict]:
             submission, options_serializer.validated_data
         )
     except RegistrationFailed:
-        formatted_tb = traceback.format_exc()
-        status = RegistrationStatuses.failed
-        result_data = {"traceback": formatted_tb}
+        if task.request.retries <= settings.CELERY_MAX_RETRIES:
+            return task.retry(
+                submission_id=submission_id,
+                countdown=3 ** task.request.retries,
+            )
+        else:
+            formatted_tb = traceback.format_exc()
+            status = RegistrationStatuses.failed
+            result_data = {"traceback": formatted_tb}
```

Comment:
Afaik it should be possible to specify all this in the task decorator, something like:

```py
@app.task(
    bind=True,
    autoretry_For=(RegistrationFailed,),
    retry_backoff=True,  # exponential backoff
    max_retries=settings.CELERY_MAX_RETRIES
)
def register_submission(task, submission_id: int) -> Optional[dict]:

```

See the docs: https://docs.celeryproject.org/en/stable/userguide/tasks.html#Task.autoretry_for

That should in turn make this code more like this:

```py
except RegistrationFailed:
    formatted_tb = traceback.format_exc()
    submission.registration_status = RegistrationStatuses.failed
    submission.registration_result = {"traceback": formatted_tb}
    submission.save(
        update_fields=[
            "registration_status",
            "registration_result",
        ]
    )
    raise
```

Even if something is going into the retry, I'd like to see the traceback in the admin interface without having to check Flower.